File: uni/xyz/adapters/windows/UniXyzCommandGateway.cs
namespace DrMarchand.UniXyz;

public sealed record UniXyzCommand(string Id, string Intent);

public interface IUniXyzCommandGateway
{
    Task SubmitAsync(UniXyzCommand command, CancellationToken cancellationToken = default);
}

public sealed class UniXyzViewModel(IUniXyzCommandGateway gateway)
{
    public Task SubmitAsync(UniXyzCommand command, CancellationToken cancellationToken = default)
    {
        // Windows submits a command through the governed gateway; no direct authoritative mutation.
        return gateway.SubmitAsync(command, cancellationToken);
    }
}

File: uni/xyz/adapters/android/UniXyzCommandGateway.kt
data class UniXyzCommand(val id: String, val intent: String)

interface UniXyzCommandGateway {
    suspend fun submit(command: UniXyzCommand)
}

class UniXyzViewModel(private val gateway: UniXyzCommandGateway) {
    suspend fun submit(command: UniXyzCommand) {
        // Android submits a command through the governed gateway; no direct authoritative mutation.
        gateway.submit(command)
    }
}
